### ISSUE TITLE
Restore with node-label flag does not work

### ISSUE THREAD

**Issue by saltenhub** [CrunchyData/postgres-operator#1206]
**Describe the bug**
When executing `pgo restore` with the `--node-label` flag, the running cluster is terminated, but no new primary is scheduled.

**To Reproduce**
1. Create cluster
2. Wait for create and backup task to finish
3. `pgo restore clustername --node-label value=key` (I tested with different kubernetes.io/hostname flag than the previous primary had)
4. See that primary and replicas are deleted
5. Wait for new primary to be scheduled on a node matching the supplied node-label
6. No new primary is scheduled

I am happy to provide additional logs/information if needed.

**Expected behavior**
A new primary to be scheduled according to the supplied `node-label`

* Operating System: Linux
* Where is this running ( Local, Cloud Provider): Local
* Storage being used (NFS, Hostpath, Gluster, etc): Hostpath
* Container Image Tag: centos7-12.1-4.2.0
* PostgreSQL Version: 12.1
* Platform (Docker, Kubernetes, OpenShift): Kubernetes
* Platform Version: 1.16.3

**Comment by jkatz** [comment]
Hi @saltenhub,

What are you trying to accomplish? Are you trying to create a new cluster on a different node? Are you trying to move an existing cluster from one node to another?

Thanks!

**Comment by saltenhub** [comment]
Hi @jkatz, thanks for the quick reply!
I was getting to that point because of issue https://github.com/CrunchyData/postgres-operator/issues/1201. After not being able to delete the previous (i.e. before manual failover) master node after manual failover, I tried restoring from backup in order to get rid of the non-removable replica. So yes, I was trying to move an existing cluster to another node.

**Comment by andrewlecuyer** [comment]
@saltenhub, this appears to be a defect in the way node labels are being applied to the restored primary Deployment.  We will therefore look at addressing this issue in an upcoming release.

Also, being that the issue in #1201 led you here per your comment above, I did want to point you to a workaround I just posted for that issue, which will hopefully help you with your specific use case: 
https://github.com/CrunchyData/postgres-operator/issues/1201#issuecomment-574334957.
